File: Hero.java
import java.util.Random;

public class Hero extends Entity {

    private static final int MAX_HEAL = 4;
    private int healCount = MAX_HEAL;
    public Hero(String name, int attack, int defence, int maxHealth, int minDamage, int maxDamage) {
        super(name, attack, defence, maxHealth, minDamage, maxDamage);
    }

    public boolean heal() {
        if (healCount-- != 0) {
            int maxHeal = (int) (0.3 * getMaxHealth());
            int healAmount = new Random().nextInt(maxHeal) + 1;
            setHealth(getHealth() + healAmount);
            return true;
        }
        return false;
    }

    public void printStats() {
        System.out.println(isAlive() ? "You, brave " + name + ", have " + getHealth() + " health points left" : "This dungeon has swallowed you, no one will find your bones");
    }
}

File: Hero.kt
class Hero(name: String?, attack: Int, defence: Int, maxHealth: Int, minDamage: Int, maxDamage: Int) : Entity(name!!, attack, defence, maxHealth, minDamage, maxDamage) {
    private var healCount = MAX_HEAL
    fun heal(): Boolean {
        if (healCount-- != 0) {
            val maxHeal = (0.3 * maxHealth).toInt()
            val healAmount = (1 ..maxHeal).random() + 1
            setHealth(getHealth() + healAmount)
            return true
        }
        return false
    }

    fun printStats() {
        println(if (isAlive) "You, brave " + name + ", have " + getHealth() + " health points left" else "This dungeon has swallowed you, no one will find your bones")
    }

    companion object {
        private const val MAX_HEAL = 4
    }
}
